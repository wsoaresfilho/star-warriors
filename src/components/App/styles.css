.app-container {
  max-width: var(--container-max-width);
  margin: auto;
  padding: 110px 10px 60px;
  height: calc(100vh - 190px);
  margin-bottom: 20px;
}

.app {
  height: 100vh;
}

.area-container {
  -webkit-box-shadow: 0px 0px 10px 0px rgba(0, 0, 0, 0.5);
  -moz-box-shadow: 0px 0px 10px 0px rgba(0, 0, 0, 0.5);
  box-shadow: 0px 0px 10px 0px rgba(0, 0, 0, 0.5);
  background-color: var(--light-bg-color-secondary);
  padding: 20px;
}

.dark {
  background-color: var(--dark-bg-color);
}

.dark .area-container {
  background-color: var(--dark-bg-color-secondary);
}
