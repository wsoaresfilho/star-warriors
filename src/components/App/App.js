import React, { PureComponent } from 'react';
import classNames from 'classnames';
import Header from '../Header/Header';
import './styles.css';

const themes = {
    LIGHT: 'light',
    DARK: 'dark',
};

class App extends PureComponent {
    constructor(props) {
        super(props);

        this.state = {
            userTheme: themes.LIGHT,
        };
    }

    isLightTheme() {
        const { userTheme } = this.state;
        return userTheme === themes.LIGHT;
    }

    render() {
        const { userTheme } = this.state;
        const appClassnames = classNames({
            app: true,
            light: userTheme === themes.LIGHT,
            dark: userTheme === themes.DARK,
        });
        const logo = this.isLightTheme ? 'logo-yellow.svg' : 'logo-black.svg';

        return (
            <div className={appClassnames}>
                <Header logo={logo} />
                {userTheme !== '' && (
                    <div className='app-container'>Container</div>
                )}
            </div>
        );
    }
}

export default App;
