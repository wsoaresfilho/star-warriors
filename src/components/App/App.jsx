import React, { PureComponent } from 'react';
import classNames from 'classnames';
import Header from '../Header/Header';
import Switch from '../Switch/Switch';
import './styles.css';

const themes = {
    LIGHT: 'light',
    DARK: 'dark',
};

class App extends PureComponent {
    constructor(props) {
        super(props);

        this.state = {
            userTheme: themes.LIGHT,
        };

        this.handleThemeChange = this.handleThemeChange.bind(this);
    }

    isLightTheme() {
        const { userTheme } = this.state;
        return userTheme === themes.LIGHT;
    }

    handleThemeChange() {
        const userTheme = this.isLightTheme() ? themes.DARK : themes.LIGHT;
        this.setState({
            userTheme,
        });
    }

    render() {
        const { userTheme } = this.state;
        const appClassnames = classNames({
            app: true,
            light: userTheme === themes.LIGHT,
            dark: userTheme === themes.DARK,
        });
        const logo = this.isLightTheme() ? 'logo-yellow.svg' : 'logo-black.svg';

        return (
            <div className={appClassnames}>
                <Header logo={logo} />
                {userTheme !== '' && (
                    <div className='app-container'>
                        <div className='theme-switch'>
                            <Switch
                                onChange={this.handleThemeChange}
                                isChecked={!this.isLightTheme()}
                            />
                        </div>
                    </div>
                )}
            </div>
        );
    }
}

export default App;
