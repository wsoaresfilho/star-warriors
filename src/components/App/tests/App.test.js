import React from 'react';
import { mount } from 'enzyme';
import App from '../App';

describe('App component', () => {
    let wrapper;

    beforeAll(() => {
        wrapper = mount(<App />);
    });

    describe('when initializes should render', () => {
        it('app element', () => {
            expect(wrapper.find('.app.light').length).toBe(1);
        });

        it('Header component', () => {
            expect(wrapper.find('Header').length).toBe(1);
        });

        it('app-container element', () => {
            expect(wrapper.find('.app-container').length).toBe(1);
        });

        it('CardsContainer component', () => {
            expect(wrapper.find('CardsContainer').exists()).toBeTruthy();
        });

        it('theme-switch element', () => {
            expect(wrapper.find('.theme-switch').length).toBe(1);
        });

        it('CustomSwitch component', () => {
            expect(wrapper.find('CustomSwitch').length).toBe(1);
        });

        it('snapshot correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('when changes to dark theme', () => {
        it('should render', () => {
            const instance = wrapper.instance();
            instance.handleThemeChange();
            wrapper.update();
            expect(wrapper.exists('.app.dark')).toBeTruthy();
        });

        it('snapshot correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
