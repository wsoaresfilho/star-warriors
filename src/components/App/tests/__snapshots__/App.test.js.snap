// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component when changes to dark theme snapshot correctly 1`] = `ReactWrapper {}`;

exports[`App component when initializes should render snapshot correctly 1`] = `ReactWrapper {}`;
